packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -416,53 +415,33 @@ export class SeedlessOnboardingController<EncryptionKey> extends BaseController<
         authKeyPair = keysFromVault.toprfAuthKeyPair;
       }
 
-      try {
-        const performFetch = async (): Promise<
-          Record<SecretType, Uint8Array[]>
-        > => {
-          const secretData = await this.toprfClient.fetchAllSecretDataItems({
-            decKey: encKey,
-            authKeyPair,
-          });
-
-          if (secretData?.length > 0 && password) {
-            // if password is provided, we need to create a new vault with the auth data. (supposedly the user is trying to rehydrate the wallet)
-            await this.#createNewVaultWithAuthData({
-              password,
-              rawToprfEncryptionKey: encKey,
-              rawToprfPwEncryptionKey: pwEncKey,
-              rawToprfAuthKeyPair: authKeyPair,
-            });
-
-            this.#persistAuthPubKey({
-              authPubKey: authKeyPair.pk,
-            });
-          }
+      const performFetch = async (): Promise<SecretMetadata[]> => {
+        const secrets = await this.#fetchAllSecretDataFromMetadataStore(
+          encKey,
+          authKeyPair,
+        );
 
-          const result: Record<SecretType, Uint8Array[]> = {
-            mnemonic: [],
-            privateKey: [],
-          };
-          const secrets =
-            SecretMetadata.parseSecretsFromMetadataStore(secretData);
+        if (password) {
+          // if password is provided, we need to create a new vault with the auth data. (supposedly the user is trying to rehydrate the wallet)
+          await this.#createNewVaultWithAuthData({
+            password,
+            rawToprfEncryptionKey: encKey,
+            rawToprfPwEncryptionKey: pwEncKey,
+            rawToprfAuthKeyPair: authKeyPair,
+          });
 
-          secrets.forEach((secret) => {
-            result[secret.type].push(secret.data);
+          this.#persistAuthPubKey({
+            authPubKey: authKeyPair.pk,
           });
```

Comment:
could this be moved into `#createNewVaultWithData`? if you create a new vault, then the corresponding auth pub key should always be set?